src/components/authpage/SignUpPage.jsx

```diff
@@ -0,0 +1,131 @@
+        }
+
+        if (!acceptTerms) {
+            setTermsError(true);
+            return;
+        } else {
+            setTermsError(false);
+        }
+
+        signUpUser(email, password);
+    };
+
+    return (
+        <main className="grand-main">
+            <div className="flex justify-content-center align-items-center h-100">
+                <div className="auth-container ">
+                    <h2 className="text-align-center font-black">Signup</h2>
+                    <div className="input-wrapper p-y-1 w-100">
+                        <label htmlFor="email" className="input-label fw-600">
+                            Email:
+                        </label>
+                        <input
+                            id="email"
+                            type="email"
+                            value={email}
+                            className={
+                                "input-field" +
+                                (emailError ? " input-color-error" : " ")
+                            }
+                            onChange={(e) => setEmail(e.target.value)}
+                        />
+                        {emailError && (
+                            <span className="input-message">
+                                Please enter valid email
+                            </span>
+                        )}
+                    </div>
+                    <div className="input-wrapper p-y-1 w-100">
+                        <label htmlFor="password" className="input-label fw-600">
+                            Password:
+                        </label>
+                        <input
+                            id="password"
+                            type="password"
+                            value={password}
+                            className={
+                                "input-field" +
+                                (passwordError ? " input-color-error" : " ")
+                            }
+                            onChange={(e) => setPassword(e.target.value)}
+                        />
+                        {passwordError && (
+                            <span className="input-message">
+                                Please enter password more that 8 characters
+                            </span>
+                        )}
+                    </div>
+                    <div className="flex justify-content-between w-100 p-y-1">
+                        <label className="font-small fw-500 flex align-items-center">
+                            <input
+                                type="checkbox"
+                                name="show-password"
+                                checked={acceptTerms}
+                                onChange={() => setAcceptTerms((s) => !s)}
+                            />{" "}
+                            <span className="p-0_5"> </span> I accept all Terms
+                            and Conditions
+                        </label>
```

Comment:
you can put this whole container in a form tag and use required in every field you this is important

Reply:
Actually I have implemented other way round the required fields get checked manually and prompts error on missing fields. I have done this provide my own custom css rather than required fields.